pyogrio/tests/test_geopandas_io.py

```diff
@@ -186,6 +186,79 @@ def test_read_fids_force_2d(test_fgdb_vsi):
         assert not df.iloc[0].geometry.has_z
 
 
+#@pytest.mark.filterwarnings("ignore: Layer")
+def test_read_sql(naturalearth_lowres):
+    # empty filter should return full set of records
+    df = read_dataframe(naturalearth_lowres, sql=None)
+    assert len(df) == 177
+    assert len(df.columns) == 6
+
+    # should return singular item
+    sql = "SELECT * FROM naturalearth_lowres WHERE iso_a3 = 'CAN'"
+    df = read_dataframe(naturalearth_lowres, sql=sql)
+    assert len(df) == 1
+    assert len(df.columns) == 6
+    assert df.iloc[0].iso_a3 == "CAN"
+    area_canada = df.iloc[0].geometry.area
+
+    # use spatialite function (needs SQLITE dialect)
+    sql = """SELECT ST_Buffer(geometry, 5) AS geometry, name, pop_est, iso_a3 
+               FROM naturalearth_lowres 
+              WHERE ISO_A3 = 'CAN'"""
+    df = read_dataframe(naturalearth_lowres, sql=sql, sql_dialect="SQLITE")
+    assert len(df) == 1
+    assert df.iloc[0].geometry.area > area_canada
+
+    # the geometry column doesn't need/cannot be specified when using the 
+    # default OGRSQL dialect but is returned nonetheless, so 4 columns
+    sql = """SELECT name, pop_est, iso_a3  
+               FROM naturalearth_lowres 
+              WHERE iso_a3 IN ('CAN', 'USA', 'MEX')
+              ORDER BY name"""
+    df = read_dataframe(naturalearth_lowres, sql=sql)
+    assert len(df.columns) == 4
+    assert len(set(df.iso_a3.unique()).difference(["CAN", "USA", "MEX"])) == 0
+    names_found = df['name'].to_list()
+    #assert len(names_found) == 4
+
+    # the geometry column doesn't need/cannot be specified when using the 
+    # default OGRSQL dialect but is returned nonetheless, so 4 columns
+    sql = """SELECT name, pop_est, iso_a3  
+               FROM naturalearth_lowres 
+              WHERE iso_a3 IN ('CAN', 'USA', 'MEX')
+              ORDER BY name"""
+    df = read_dataframe(naturalearth_lowres, sql=sql, skip_features=1, max_features=1)
+    assert len(df.columns) == 4
+    assert len(set(df.iso_a3.unique()).difference(["CAN", "USA", "MEX"])) == 0
```

Comment:
Replace lines 232-235 with 

```Python
    assert len(df) == 1
    assert df.iso_a3.tolist() == ["MEX"]
```

Suggest adding another couple tests that compounds the sql `LIMIT` with `skip_features` / `max_features`:

```Python
    sql = "SELECT name, pop_est, iso_a3 FROM naturalearth_lowres LIMIT 1"
    with pytest.raises(ValueError, match="'skip_features' must be between 0 and 0"):
        df = read_dataframe(naturalearth_lowres, sql=sql, skip_features=1)
```
(can't skip when there is only 1 result)

```Python
sql = "SELECT name, pop_est, iso_a3 FROM naturalearth_lowres LIMIT 1"
    df = read_dataframe(naturalearth_lowres, sql=sql, max_features=3)
    assert len(df) == 1
```